server/core/src/irods_resource_manager.cpp

```diff
@@ -444,497 +405,575 @@ namespace irods
+            log_resc::error("Invalid parent resource id [{}] for resource [{}].", parent_id, _resc_name);
+            return;
+        }
 
-        return SUCCESS();
+        resource_ptr parent_resc = resource_id_map_[parent_id];
 
-    } // init_child_map
-
-// =-=-=-=-=-=-=-
-// public - print the list of local resources out to stderr
-    void resource_manager::print_local_resources() {
-        lookup_table< boost::shared_ptr< resource > >::iterator itr;
-        for ( itr = resource_name_map_.begin(); itr != resource_name_map_.end(); ++itr ) {
-            std::string loc, path, name;
-            error path_err = itr->second->get_property< std::string >( RESOURCE_PATH, path );
-            error loc_err  = itr->second->get_property< std::string >( RESOURCE_LOCATION, loc );
-            if ( path_err.ok() && loc_err.ok() && "localhost" == loc ) {
-                rodsLog( LOG_NOTICE, "   RescName: %s, VaultPath: %s\n",
-                         itr->first.c_str(), path.c_str() );
+        std::string parent_child_context;
+        _resc_ptr->get_property<std::string>(RESOURCE_PARENT_CONTEXT, parent_child_context);
+
+        parent_resc->add_child(std::string{_resc_name}, parent_child_context, _resc_ptr);
+        _resc_ptr->set_parent(parent_resc);
+
+        log_resc::debug("{} - add [{}][{}] to [{}]", __func__, _resc_name, parent_child_context, parent_id);
+    } // init_parent_child_relationship_impl
+
+    void resource_manager::add_child_to_resource(rodsLong_t _parent_resc_id, rodsLong_t _child_resc_id)
+    {
+        auto parent_resc_ptr = resource_id_map_[_parent_resc_id];
+        std::string resc_name;
+        if (const auto err = parent_resc_ptr->get_property<std::string>(RESOURCE_NAME, resc_name); !err.ok()) {
+            THROW(err.code(), err.result());
+        }
+
+        auto child_resc_ptr = resource_id_map_[_child_resc_id];
+        std::string child_resc_name;
+        if (const auto err = child_resc_ptr->get_property<std::string>(RESOURCE_NAME, resc_name); !err.ok()) {
+            THROW(err.code(), err.result());
+        }
+
+        std::string parent_child_context;
+        auto err = child_resc_ptr->get_property<std::string>(RESOURCE_PARENT_CONTEXT, parent_child_context);
+        log_resc::info(err.result());
+
+        err = parent_resc_ptr->add_child(std::string{child_resc_name}, parent_child_context, child_resc_ptr); // TODO Can throw!!!
+        log_resc::info(err.result());
+        err = child_resc_ptr->set_parent(parent_resc_ptr); // TODO Can throw!!!
+        log_resc::info(err.result());
+        err = child_resc_ptr->set_property<std::string>(RESOURCE_PARENT, std::to_string(_parent_resc_id)); // TODO Can throw!!!
+        log_resc::info(err.result());
+    } // add_child_to_resource
+
+    void resource_manager::add_child_to_resource(
+        const std::string_view _parent_resc_name,
+        const std::string_view _child_resc_name)
+    {
+        log_resc::info(">>>>>>>>>>>>> IN {}", __func__);
```

Comment:
ew.

Reply:
Yes, a known leftover from debugging. These will not be part of the final PR.

Reply:
Done